app/models/embedded_ansible_worker/runner.rb

```diff
@@ -57,6 +52,25 @@ def update_embedded_ansible_provider
     provider.authentication_check
   end
 
+  def synchronize_provider_with_server
+    server = MiqServer.my_server(true)
+
+    provider.name = "Embedded Ansible"
+    provider.zone = server.zone
+    provider.url  = provider_url
+    provider.verify_ssl = 0
+
+    provider.save!
+  end
+
+  def provider_matches_server?
```

Comment:
I think we could probably name this better as it's handling both the time aspect as well as the matching.

Unfortunately the best I could come up with was `synchronize_provider_with_server?`